src/charm.py

```diff
@@ -666,12 +666,15 @@ def _create_upf_configuration_file(self) -> tuple[bool, bool]:
         restart = False
         recreate_pod = False
         core_ip_address = self._get_network_ip_config(CORE_INTERFACE_NAME)
+        core_ip_masquerade_flag = self._charm_config.core_ip_masquerade
         content = render_bessd_config_file(
             upf_hostname=self._upf_hostname,
             upf_mode=self._charm_config.upf_mode,
             access_interface_name=ACCESS_INTERFACE_NAME,
             core_interface_name=CORE_INTERFACE_NAME,
-            core_ip_address=core_ip_address.split("/")[0] if core_ip_address else "",
+            core_ip_masquerade=core_ip_address.split("/")[0]
```

Comment:
```suggestion
            core_ip_address=core_ip_address.split("/")[0]
```

I would rename this to `core_ip_address` for clarity. It is what it contains after all.

Reply:
Ok I addressed this. I initially went for `core_ip_masquerade` because this is the variable used in the json configuration file.